Bio stats assignment.py

```diff
@@ -0,0 +1,143 @@
+
+    # O(1) + O(1) + O(N*(N+1)) complexity uptil now
+
+    data_dict = dict()  # O(1).  we create empty dictionary
+    # we loop over the index of each element in a list (columns on the data)
+
+    for i in range(len(data_lst[1])):  # O(N).
+        values = []  # create a list of values O(1)
+        for row in range(len(data_lst)):  # loop over each row.  O(N)
+            if row == 0:  # check if its a header row O(1)
+                key = data_lst[row][i]  # O(1) # create key variable as a column name
+            else:  # either of the two operation performed
+                values.append(data_lst[row][i])  # O(1)
+
+            # check condition O(1) and performing required operation O(1)
+
+        # append to values if not a column name
+        # our final dictionary will have column names as keys and values will be
+        # a list of all the elements corresponding to a column name for example
+        # Patient ID is column name and a list of all patient IDs is the value of that key
+        data_dict[key] = values  # insert key and value into dictionary
+    # O(1) +O(N((1+N(1+1)) +1))
+    return data_dict  # O(1)
+
+
+# complexity of parse data function
+# O(1) + O(1) + O(N*(N+1)) + O(1) +O(N((1+N(1+1)) +1)) + O(1)
+# O(4) + O(N^2 + N) +  O(2N^2 + 2N)
+# O(3N^2 + 3N + 4) ->  `O(N^2)`
+
+
+def num_older_than(age: Union[int, float], patient_data: dict[str, list[Any]]) -> int:
+    """Return number of patients greater than age."""
+    today_date: datetime = datetime.today()  # O(1) generate todays date
+
+    patient_data["PatientDateOfBirth"] = [
+        datetime.strptime(i, "%Y-%m-%d %H:%M:%S.%f")
+        for i in patient_data["PatientDateOfBirth"]  # list comprehension
+    ]  # we just convert every date to datetime O(N) method and make a
+    # list if it O(N). Insert this list as value to our
+    # data dictionary  O(N+N+1) -> O(2N+ 1) -> O(N)
+
+    patient_data["Age"] = [
+        ((today_date - i).days / 365.25) for i in patient_data["PatientDateOfBirth"]
+    ]
+    # in the above list comprehension
+    # we insert another value in dictionary,O(1)
+    # make a list. Loop over birthdate of patients and subtract the todays date O(N)
+    # extract days from the difference O(N)
+    #  divide by 365.25 (0.25 for leap year) O(N)
+    # append these values to the list.
+    # O(N+N+N+N) -> O(4N) -> O(N)
+
+    counter: int = 0  # create a new variable to count the number greater than age. O(1)
+    for i in patient_data["Age"]:  # We loop over age and see if the age if
+        if i > age:  # a number is greater than the given value
+            counter += 1  # we add 1 to the value of counter and
+            # than assign it to to the counter
+
+    # complexity of the above loop O(N(1+2)) -> O(3N) -> O(N)
+
+    return counter  # O(1)
+
+
+# complexity of num_older_than function is
+# O(N) + O(N) + O(N) + O(2) -> O(N)
+
+
+def sick_patients(
+    lab: str, gt_lt: str, value: float, data: dict[str, list[Any]]
```

Comment:
Is this `data` the patients table or labs table? Also clarify the type hint here and below.